.pre-commit-config.yaml

```diff
@@ -53,6 +54,12 @@ repos:
       language: system
       entry: shellcheck --severity=warning -e SC2068
       types: [shell]
+    - id: run-tests
+      name: run tests before pushing
+      language: system
+      entry: ninja -C build test
```

Comment:
I think this also means needing to run "meson build" first no?

I also worry about the default configuration not working for some environment that might be pushing.

Reply:
> Haven't tried yet but how much time does this seem to add?

Depends of your state. From a clean repo, on my machine, meson takes 4 seconds and ninja 53 seconds. On an already built repo where only the tests need to execute, 9 seconds.

> It's possible to skip checks from pre-commit, so small obvious things we could be setting that environment variable.

My main issue is that there's no visual feedback while the tests are running: there is the "pre-commit: run tests" line, but no sense of progress of the build+test process itself.

Reply:
> I think this also means needing to run "meson build" first no?

Done.

> I also worry about the default configuration not working for some environment that might be pushing.

Done.

Reply:
> My main issue is that there's no visual feedback while the tests are running: there is the "pre-commit: run tests" line, but no sense of progress of the build+test process itself.

The lack of visual feedback is "generally" too problematic IMO.  I think if we add this is should be a "default" N in `contrib/setup` and should clearly disclaim that it will slow down pushes.

Reply:
The latest version defaults to N, says so explicitly, has a detailed explanation of consequences, and shouldn't impact the CI at all (unless I missed something). I'm also happy to drop this PR if you don't think it's necessary.